### DISCUSSION TITLE
KPointsAndWeights has been ignored by the parser

### DISCUSSION THREAD

**Discussion by qixiangyin** [dftbplus/dftbplus#1354] [Q&A] [answered]
Dear  developers.
I am a beginner in DFTBPlus.I have encountered a problem with band structure calculations.When I run the input file dftb_in.hsd, it always reminds me that KPointsAndWeights is ignored.
Here are the details. 
input：
Geometry = xyzFormat {
<<<"geom4.xyz"
}

Hamiltonian = DFTB {
  Scc = Yes
  SccTolerance = 0.1
  SlaterKosterFiles {
    S-S = "../../slakos/mio-ext/S-S.skf"
    Ti-Ti = "../../slakos/mio-ext/Ti-Ti.skf"
    S-Ti = "../../slakos/mio-ext/S-Ti.skf"
    Ti-S = "../../slakos/mio-ext/Ti-S.skf"
  }
  MaxAngularMomentum {
    Ti = "d"
    S = "p"
  }

  KPointsAndWeights = SupercellFolding {
    4 0 0
    0 4 0
    0 0 1
    0.5 0.5 0.0
  }
}

Options {}

Analysis {
  ProjectStates {
    Region {
      Atoms = Ti
      ShellResolved = Yes
      Label = "dos_ti"
    }
    Region {
      Atoms = S
      ShellResolved = Yes
      Label = "dos_s"
     }
  }  
}

ParserOptions {
  ParserVersion = 12
}

output:
ERROR!
-> Code halting due to the presence of errors in dftb in file.ERROR STOP
Done.
WARNING!
-> The following l node(s) have been ignored by the parser:(1)
Path: dftbplusinput/Hamiltonian/DFTB/KPointsAndWeightsLine: 20-25 (File: dftb in.hsd)
ERROR!
-> Code halting due to the presence of errors in dftb in file.
ERROR STOP

Thanks in advance

**Comment by aradi** [comment, marked as answer]
The problem seems to be, that your structure is not periodic and therefore DFTB+ tries to do a non-periodic calculation, where no K-Points are needed. Note, the XYZ-format does not contain any lattice information. You the VASP or the GEN-format instead, which contain also the supercell vectors.

**Reply by qixiangyin** [reply]
Thank you very much for your answer. The problem has been solved.
